backup: TestFullClusterRestoreWithUserIDs failed

backup.TestFullClusterRestoreWithUserIDs [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20510705?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20510705?buildTab=artifacts#/) on master @ [3476899d48145434a9ea53f95c537e670927bd26](https://github.com/cockroachdb/cockroach/commits/3476899d48145434a9ea53f95c537e670927bd26):


```
=== RUN   TestFullClusterRestoreWithUserIDs
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/29162597e3744e49f10d1d09f33805c2/logTestFullClusterRestoreWithUserIDs2115485594
    test_log_scope.go:82: use -show-logs to present logs inline
    full_cluster_backup_restore_test.go:1192: query 'SELECT username, "hashedPassword", "isRole", user_id FROM system.users ORDER BY user_id': expected:
        root, , false, 1
        admin, , true, 2
        test1, NULL, false, 100
        test2, NULL, false, 101
        
        got:
        root, , false, 1
        admin, , true, 2
        test1, NULL, false, 100
        test2, NULL, false, 102
        
    testutils.go:307: no Invalid Descriptors
    testutils.go:307: no Invalid Descriptors
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/29162597e3744e49f10d1d09f33805c2/logTestFullClusterRestoreWithUserIDs2115485594
--- FAIL: TestFullClusterRestoreWithUserIDs (3.98s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFullClusterRestoreWithUserIDs.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-54719

backup.TestFullClusterRestoreWithUserIDs [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20530362?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20530362?buildTab=artifacts#/) on master @ [ac68bcaf4a547b9d24127a6da8ee01df7753fea1](https://github.com/cockroachdb/cockroach/commits/ac68bcaf4a547b9d24127a6da8ee01df7753fea1):


```
=== RUN   TestFullClusterRestoreWithUserIDs
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/29162597e3744e49f10d1d09f33805c2/logTestFullClusterRestoreWithUserIDs2942352481
    test_log_scope.go:82: use -show-logs to present logs inline
    full_cluster_backup_restore_test.go:1192: query 'SELECT username, "hashedPassword", "isRole", user_id FROM system.users ORDER BY user_id': expected:
        root, , false, 1
        admin, , true, 2
        test1, NULL, false, 100
        test2, NULL, false, 101
        
        got:
        root, , false, 1
        admin, , true, 2
        test1, NULL, false, 100
        test2, NULL, false, 102
        
    testutils.go:307: no Invalid Descriptors
    testutils.go:307: no Invalid Descriptors
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/29162597e3744e49f10d1d09f33805c2/logTestFullClusterRestoreWithUserIDs2942352481
--- FAIL: TestFullClusterRestoreWithUserIDs (2.49s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFullClusterRestoreWithUserIDs.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/backup/full_cluster_backup_restore_test.go
+++ b/pkg/backup/full_cluster_backup_restore_test.go
@@ -1189,29 +1189,29 @@ func TestFullClusterRestoreWithUserIDs(t *testing.T) {
 	sqlDB.Exec(t, `CREATE USER test2`)
 	sqlDB.Exec(t, `BACKUP INTO $1`, localFoo)
 
-	sqlDB.CheckQueryResults(t, `SELECT username, "hashedPassword", "isRole", user_id FROM system.users ORDER BY user_id`, [][]string{
-		{"root", "", "false", "1"},
-		{"admin", "", "true", "2"},
-		{"test1", "NULL", "false", "100"},
-		{"test2", "NULL", "false", "101"},
+	sqlDB.CheckQueryResults(t, `SELECT username, "hashedPassword", "isRole" FROM system.users ORDER BY user_id`, [][]string{
+		{"root", "", "false"},
+		{"admin", "", "true"},
+		{"test1", "NULL", "false"},
+		{"test2", "NULL", "false"},
 	})
 	// Ensure that the new backup succeeds.
 	sqlDBRestore.Exec(t, `RESTORE FROM LATEST IN $1`, localFoo)
 
-	sqlDBRestore.CheckQueryResults(t, `SELECT username, "hashedPassword", "isRole", user_id FROM system.users ORDER BY user_id`, [][]string{
-		{"root", "", "false", "1"},
-		{"admin", "", "true", "2"},
-		{"test1", "NULL", "false", "100"},
-		{"test2", "NULL", "false", "101"},
+	sqlDBRestore.CheckQueryResults(t, `SELECT username, "hashedPassword", "isRole" FROM system.users ORDER BY user_id`, [][]string{
+		{"root", "", "false"},
+		{"admin", "", "true"},
+		{"test1", "NULL", "false"},
+		{"test2", "NULL", "false"},
 	})
 
 	sqlDBRestore.Exec(t, `CREATE USER test3`)
 
-	sqlDBRestore.CheckQueryResults(t, `SELECT username, "hashedPassword", "isRole", user_id FROM system.users ORDER BY user_id`, [][]string{
-		{"root", "", "false", "1"},
-		{"admin", "", "true", "2"},
-		{"test1", "NULL", "false", "100"},
-		{"test2", "NULL", "false", "101"},
-		{"test3", "NULL", "false", "102"},
+	sqlDBRestore.CheckQueryResults(t, `SELECT username, "hashedPassword", "isRole" FROM system.users ORDER BY user_id`, [][]string{
+		{"root", "", "false"},
+		{"admin", "", "true"},
+		{"test1", "NULL", "false"},
+		{"test2", "NULL", "false"},
+		{"test3", "NULL", "false"},
 	})
 }

